include/deal.II/grid/grid_generator.h

```diff
@@ -768,6 +768,36 @@ namespace GridGenerator
                       const Point<dim> &  center = Point<dim>(),
                       const double        radius = 1.);
 
+  /**
+   * Generate a mesh consisting of the unit cube joined with a copy shifted by
+   * $s = (1,0,0)$. Depending on the flags passed either the right or the
+   * left cube (when looking at the positively oriented (x,z)-plane) contains a
+   * face that is either not in standard orientation and/or is rotated by either
+   * $\pi/2$, $\pi$ or $3/2\pi$.
+   *
+   * This mesh is not overly useful from the practical point of view. For
+   * debugging purposes it can be used to check for orientation issues for
+   * tensor valued finite elements.
+   *
+   * @tparam dim
+   * @param[out] tria input triangulation
+   * @param[in] face_orientation <code>true</code> if the face is the not in
+   * standard orientation
+   * @param[in] face_flip <code>true</code> if the face is rotated by +180
+   * degrees
+   * @param[in] face_rotation <code>true</code> if the face is rotated
+   * (additionally) by +90 degrees
+   * @param[in] manipulate_left_cube <code>true</code> if the left cube is
+   * re-ordered
+   */
+  template <int dim>
+  void
+  orientation_test_mesh(Triangulation<dim> &tria,
```

Comment:
How about calling it "non_standard_orientation_mesh" or simply "non_standard_mesh"?